h11/_receivebuffer.py

```diff
@@ -79,21 +90,29 @@ def maybe_extract_at_most(self, count):
         self._start += len(out)
         return out
 
-    def maybe_extract_until_next(self, needle):
+    def maybe_extract_until_next(self, needle_regex, max_needle_length):
```

Comment:
Should this actually be `min_needle_length`?

Reply:
Ie. always resume searching from the first possible point where there could still be a match.

Reply:
Ah yeah I see it `max_needle_length`, because that pushes the restart *back* as far as possible.